### DISCUSSION TITLE
How do I run specific commands for specific projects in one git repository?

### DISCUSSION THREAD

**Discussion by neptunecentury** [evilmartians/lefthook#882] [Q&A] [answered]
I have a git repository which contains two projects. One is an asp.net core web api, and the other is a frontend vue project. They are in separate directories under the main git repo directory.

My understanding is that a hook is executed per repository, so for example, if I configure a pre-commit hooks, any file that is staged and about to be committed will be passed in {staged_files} to my configured commands. The problem is, if I want to run my linter only for the frontend vue app and not the .net core files if I have any modified. But my linter and its configuration is installed in the vue app's directory.

How can I set up lefthook config to run the linter in the vue app directory ONLY when a file from that directory is staged?
I have been searching, and I think the glob configuration may help, but I am not sure how to set it up.

Thanks.

**Comment by mrexox** [comment, marked as answer]
So, you want to run the linter only in a specific folder, right?

You can use [`root`](https://github.com/evilmartians/lefthook/blob/master/docs/configuration.md#root) option. Besides filtering by directory it also changes the `pwd` to the given directory, so if you have the following project structure, you can use `yarn lint {staged_files}` and it will be executed in the `vue` directory:
```
.
├── dot-net/
├── lefthook.yml
└── vue/
    ├── package.json
    └── yarn.lock
```

Example `lefthook.yml` config

```yml
pre-commit:
  commands:
    lint:
      root: vue # filter by directory
      glob: "*.{ts,js}" # filter by glob
      run: yarn lint {staged_files}
```
